File: UnitConverterApp/app/src/main/java/com/example/unitconverterapp/MainActivity.java
package com.example.unitconverterapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    Button button;
    TextView textView;
    EditText editText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button=findViewById(R.id.button);
        editText=findViewById(R.id.editText);
        textView=findViewById(R.id.textView);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String str=editText.getText().toString();

                textView.setText(""+converter(Double.parseDouble(str)));
            }
        });


    }

    public double converter(double kilos){
        return kilos * 2.20462262;
    }
}

File: UnitConverterApp/app/src/main/java/com/example/unitconverterapp/MainActivity.kt
package com.example.unitconverterapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val et:EditText=findViewById(R.id.editText);
        val btn:Button=findViewById(R.id.button);
        val result_txt:TextView=findViewById(R.id.textView);

        btn.setOnClickListener(){
            val str:Double =et.text.toString().toDouble();
            result_txt.setText(""+convert(str));

        }
    }

    fun convert(kilos :Double):Double{
        return kilos*2.20462262;
    }
}